server/lib/Room.js

```diff
@@ -1124,37 +1107,21 @@ class Room extends EventEmitter
 				}
 				catch (error)
 				{
-					logger.error(
+					throw new Error(
```

Comment:
@so010 have a look at this change, I think we had an error here as when the creation of the producer failed we have only logged it and the code below was working as if the producer was created. I have changed it to throw an error now

Reply:
I tested it and I can confirm that  "not iterable" error is gone, however it seems that consumer/producer metrics are not collected/exported. For some strange reason it is like "workers" map (in lib/stats/metrics/default.js) is sometimes empty.

Reply:
@sd4v1d I had a look at the code, and I would say it cannot be what workers is empty. How can I configure and run prometheus to reproduce?

Reply:
It is working after last commit.

Reply:
I think that's right if we throw a error here. It is caught on higher level and logged there